airbyte-config/init/src/main/java/io/airbyte/config/init/LocalDefinitionsProvider.java

```diff
@@ -48,22 +48,26 @@ public LocalDefinitionsProvider(final Class<?> seedResourceClass) throws IOExcep
     this.seedResourceClass = seedResourceClass;
 
     // TODO remove this call once dependency injection framework manages object creation
-    initialize();
+    loadDefinitions();
   }
 
   // TODO will be called automatically by the dependency injection framework on object creation
-  public void initialize() throws IOException {
-    this.sourceDefinitions =
-        parseDefinitions(this.seedResourceClass, SeedType.STANDARD_SOURCE_DEFINITION.getResourcePath(), SeedType.SOURCE_SPEC.getResourcePath(),
-            SeedType.STANDARD_SOURCE_DEFINITION.getIdName(), SeedType.SOURCE_SPEC.getIdName(), StandardSourceDefinition.class);
-    this.destinationDefinitions = parseDefinitions(this.seedResourceClass, SeedType.STANDARD_DESTINATION_DEFINITION.getResourcePath(),
-        SeedType.DESTINATION_SPEC.getResourcePath(), SeedType.STANDARD_DESTINATION_DEFINITION.getIdName(), SeedType.DESTINATION_SPEC.getIdName(),
-        StandardDestinationDefinition.class);
+  public void loadDefinitions() {
```

Comment:
Since local definitions are read from the resources, we could annotate here with `@PostConstruct`